### PR TITLE
Fix aggregation by partitions

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed bug in aggregation by partitions (setting `allow_aggregate_partitions_independently`) that led to the `DB::Exception: Unknown aggregated data variant.` error.

### CODE REVIEW

**Comment by nickitat** [comment]
I suppose flaky checked failed previously because of sparse columns...

File: src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp [hunk 1]
```diff
@@ -202,7 +206,7 @@ size_t tryAggregatePartitionsIndependently(QueryPlan::Node * node, QueryPlan::No
         return 0;
 
     if (!reading->willOutputEachPartitionThroughSeparatePort()
-        && isPartitionKeySuitsGroupByKey(*reading, expression_step->getExpression(), *aggregating_step))
+        && isPartitionKeySuitsGroupByKey(*reading, expression_step->getExpression()->clone(), *aggregating_step))
```
Comment (devcrafter) [inline src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp:209]: Not sure if `clone()` is necessary here. I don't see it's changing inside `isPartitionKeySuitsGroupByKey`

**Review by devcrafter — APPROVED** [review]
LGTM with comment

File: src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp (hunk 1 above)
Comment (nickitat) [reply src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp:209]: `removeUnusedActions` modifies actions dag

**Comment by nickitat** [comment]
Stateless tests (aarch64) — 
02435_rollback_cancelled_queries | FLAKY 
-- | --

**Pull request merged**